content_scripts/normal.js

```diff
@@ -415,8 +415,12 @@ function createNormal() {
         Mode.showStatus();
         if (runtime.conf.stealFocusOnLoad && !isInUIFrame()) {
             var elm = getRealEdit(event);
+            // when in insert/passthrough mode, allow page to grab focus normally
+            // However, we still clear passFocus for future clicks.
+            const typing_mode = Mode.stack()[0] === PassThrough
```

Comment:
The logic for each mode should be placed within its own mode, for this case specifically, please follow https://github.com/brookhong/Surfingkeys/blob/master/content_scripts/normal.js#L331-L333

and try to addEventListener for `focus`. 

For `Insert` mode, we already have some handling for the case, please see https://github.com/brookhong/Surfingkeys/blob/master/content_scripts/insert.js#L370